test-vite-app-v2/src/components/__tests__/load-public-asset.cy.ts

```diff
@@ -0,0 +1,12 @@
+import LoadAsset from '../LoadPublicAsset.vue';
+
+describe('load assets', () => {
+  it('verifies that the image is loaded and rendered correctly', () => {
+    cy.mount(LoadAsset);
+    cy.dataCy('test-image')
+      .should('be.visible')
+      .and(($img) => {
+        expect(($img[0] as HTMLImageElement).naturalWidth).to.be.greaterThan(0);
+      });
```

Comment:
Alternative ways as per https://glebbahmutov.com/cypress-examples/recipes/image-loaded.html

```suggestion
      .and('have.prop', 'naturalWidth')
      .should('be.greaterThan', 0);
```
```suggestion
      .invoke('prop', 'naturalWidth')
      .should('be.greaterThan', 0);
```

I don't have a strong preference, just wanted to share.

Reply:
The versions proposed by Yusuf, in this case, are a better choice
They avoid a manual TS casting, and thus reduce the difference between JS and TS scaffolding

I won't change them right now, but if we touch again these files we should update them